serving/docker/partition/sm_neo_dispatcher.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+#
+# Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file
+# except in compliance with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS"
+# BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+import subprocess
+import sys
+import os
+from enum import Enum
+
+from sm_neo_utils import get_neo_env_vars
+from utils import update_kwargs_with_env_vars, load_properties
+
+VALID_LOAD_FORMATS = ["sagemaker_fast_model_loader"]
+
+
+class NeoTask(Enum):
+    """
+    Enum representing available Neo optimization tasks, including their names and script paths
+    """
+
+    TENSORRT_LLM = ("TensorRT-LLM compilation",
+                    "/opt/djl/partition/sm_neo_trt_llm_partition.py")
+    NEURON = ("Neuron compilation",
+              "/opt/djl/partition/sm_neo_neuron_partition.py")
+    QUANTIZATION = ("Quantization", "/opt/djl/partition/sm_neo_quantize.py")
+    SHARDING = ("SageMaker Fast Model Loader sharding",
+                "/opt/djl/partition/sm_neo_shard.py")
+
+    def __init__(self, task_name: str, script_path: str):
+        self.task_name = task_name
+        self.script_path = script_path
+
+
+class NeoDispatcher:
+
+    def __init__(self):
+        self.serving_features = os.environ.get("SERVING_FEATURES")
+        self.properties = self._get_serving_properties()
+
+    def _get_serving_properties(self) -> dict:
+        neo_environ = get_neo_env_vars()
+        properties = update_kwargs_with_env_vars({})
+        properties.update(
+            load_properties(neo_environ["SM_NEO_INPUT_MODEL_DIR"]))
+        return properties
+
+    def is_valid_sharding_config(self):
+        # TODO: once pp is supported, add add requirement and remove default value
+        return (self.properties.get("option.load_format") in VALID_LOAD_FORMATS
+                and self.properties.get("option.tensor_parallel_degree")
+                is not None)
```

Comment:
how would this work today if someone did specify pp=2 for example? do we need to validate and prevent that specifically?

Reply:
yes Fast Model Loading is not supported with PP for this release. It is WIP. 

I will add a validation to fail fast.